src/messaging/ExchangeContext.cpp

```diff
@@ -115,55 +114,18 @@ CHIP_ERROR ExchangeContext::SendMessageImpl(Protocols::Id protocolId, uint8_t ms
                                             const SendFlags & sendFlags, Transport::PeerConnectionState * state)
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
-    PayloadHeader payloadHeader;
 
     // Don't let method get called on a freed object.
     VerifyOrDie(mExchangeMgr != nullptr && GetReferenceCount() > 0);
 
-    if (state == nullptr)
-    {
-        state = mExchangeMgr->GetSessionMgr()->GetPeerConnectionState(mSecureSession);
-    }
-
-    VerifyOrExit(state != nullptr, err = CHIP_ERROR_NOT_CONNECTED);
-
     // we hold the exchange context here in case the entity that
     // originally generated it tries to close it as a result of
     // an error arising below. at the end, we have to close it.
     Retain();
 
-    // Set the exchange ID for this header.
-    payloadHeader.SetExchangeID(mExchangeId);
-
-    // Set the protocol ID and message type for this header.
-    payloadHeader.SetMessageType(protocolId, msgType);
-
-    payloadHeader.SetInitiator(IsInitiator());
-
-    // If sending via UDP and auto-request ACK feature is enabled, automatically request an acknowledgment,
-    // UNLESS the NoAutoRequestAck send flag has been specified.
-    if ((state->GetPeerAddress().GetTransportType() == Transport::Type::kUdp) && mReliableMessageContext.AutoRequestAck() &&
-        !sendFlags.Has(SendMessageFlags::kNoAutoRequestAck))
-    {
-        payloadHeader.SetNeedsAck(true);
-    }
-
-    // If there is a pending acknowledgment piggyback it on this message.
-    if (mReliableMessageContext.HasPeerRequestedAck())
-    {
-        payloadHeader.SetAckId(mReliableMessageContext.mPendingPeerAckId);
+    ExchangeTransport::ExchangeInfo exchangeInfo = { IsInitiator(), mExchangeId };
```

Comment:
PayloadHeader is constructed by SendFlags, we don't need extra translation if we construct PayloadHeader within this function, and also can save a bunch of parameters in ExchangeTransport::SendMessage

Reply:
Same response (https://github.com/project-chip/connectedhomeip/pull/5938#discussion_r613214753) applies here.